cpp/src/IceDiscovery/LookupI.cpp

```diff
@@ -349,27 +334,48 @@ LookupI::findAdapter(const AdapterCB& cb, const std::string& adapterId)
 }
 
 void
-LookupI::foundObject(const Ice::Identity& id, const string& requestId, const Ice::ObjectPrxPtr& proxy)
+LookupI::foundObject(const Ice::Identity& id, const string& requestId, const optional<ObjectPrx>& proxy)
 {
     lock_guard lock(_mutex);
+    if (!proxy)
```

Comment:
I think it would be cleaner if this function took a non-optional proxy and we did the throwing in the function that calls this.